frontend/src/components/Login.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from "react";
+import { useMutation } from "react-query";
+import { useNavigate } from "react-router-dom";
+import {
+  TextField,
+  Button,
+  Container,
+  Typography,
+  Box,
+  Alert,
+  Link as MuiLink,
+} from "@mui/material";
+import { useForm } from "../hooks/useForm";
+import { loginSchema, LoginFormData } from "../utils/validationSchema";
+import { login as loginApi } from "../services/api";
+import { useAuth } from "../hooks/useAuth";
+import { FormData } from "../types";
+
+const Login: React.FC = () => {
+  const navigate = useNavigate();
+  const { login } = useAuth();
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({ schema: loginSchema });
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  const mutation = useMutation(loginApi, {
+    onSuccess: (data) => {
+      login(data as FormData);
+      navigate("/");
+    },
+    onError: (error: any) => {
+      setErrorMessage(error.message || "An error occurred during login");
+    },
+  });
+
+  const onSubmit = (data: LoginFormData) => {
+    setErrorMessage(null);
+    mutation.mutate(data);
+  };
+
```

Comment:
you can separate the states into a different hook. it makes the code more readable and concise.